packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -1113,6 +1216,33 @@ export default class TabAdvanced extends Component {
               individualActions={VUSB_INDIVIDUAL_ACTIONS}
             />
             <br />
+            <h3>{_('attachedPcis')}</h3>
+            <div className='text-info'>
+              <i className='d-block'>
+                <Icon icon='info' /> {_('infoUnknownPciOnNonRunningVm')}
+              </i>
+              <i className='d-block'>
+                <Icon icon='info' /> {_('attachingDetachingPciNeedVmBoot')}
+              </i>
+            </div>
+            <ActionButton
+              btnStyle='primary'
+              disabled={!_isPciPassthroughAvailable}
+              handler={this._attachPcis}
+              icon='connect'
+              tooltip={_isPciPassthroughAvailable ? undefined : _('onlyAvailableXcp83OrHighter')}
+            >
+              {_('attachPcis')}
+            </ActionButton>
+            <SortedTable
+              actions={PCI_ACTIONS}
+              collection={this._getAttachedPciIds()}
```

Comment:
Why don't we pass a collection of PCIs instead of IDs here?

Reply:
From a VM, it's not possible to retrieve the attached PCI XAPI objects because they are identified by their PCI ID (0/0000:08:00.0) rather than their XAPI ID. PCI IDs are not unique within a pool but they are unique within a host. This means that as long as the VM is not on a host, it's not possible to determine which PCI ID corresponds to which XAPI PCI.

Passing the attached IDs instead of a collection allows us to display some information even if the VM is halted, such as the number of attached PCIs and their IDs.